src/timeseriesflattener/resolve_multiple_functions.py

```diff
@@ -126,8 +127,57 @@ def concatenate(grouped_df: DataFrame) -> DataFrame:
     Returns:
         DataFrame: Dataframe with value column containing the concatenated values.
     """
+
     return grouped_df.apply(
         lambda x: Series(
             {"value": " ".join(x.value)},
         ),
     )
+
+
+@resolve_multiple_fns.register("mean_len")
+def mean_len(grouped_df: DataFrame) -> DataFrame:
+    """Returns the mean length of values. This is useful for text data.
+
+    Args:
+        grouped_df (DataFrame): A dataframe sorted by descending timestamp, grouped by citizen.
+
+    Returns:
+        DataFrame: Dataframe with value column containing the concatenated values.
+    """
+    return grouped_df.apply(
+        lambda x: Series(
+            {"value": x.value.str.len().mean()},
+        ),
+    )
+
+
+@resolve_multiple_fns.register("type_token_ratio")
+def type_token_ratio(grouped_df: DataFrame) -> DataFrame:
+    """Returns the type-token ratio. This is useful for text data.
+
+    Args:
+        grouped_df (DataFrame): A dataframe sorted by descending timestamp, grouped by citizen.
+
+    Returns:
+        DataFrame: Dataframe with value column containing the concatenated values.
+    """
+
+    return (
+        grouped_df["value"]
+        .apply(
+            lambda x: len(
+                nltk.Counter(
+                    " ".join(
+                        x.replace(r"[^ÆØÅæøåA-Za-z0-9 ]+", "", regex=True).str.lower(),
+                    ).split(" "),
+                ),
+            )
+            / len(
+                " ".join(
+                    x.replace(r"[^ÆØÅæøåA-Za-z0-9 ]+", "", regex=True).str.lower(),
+                ).split(" "),
+            ),
+        )
+        .reset_index()
```

Comment:
seems a bit excessive with all the regex'ing and the counter. Can be simplified to something like the following with minimal loss of information and significant speed up. You'd get issues with punctuation if you don't add the `.replace` though. 

Might be worth considering making these features post-hoc though. [TextDescriptives](https://github.com/HLasse/TextDescriptives/) can calculate many of these common features at once, so perhaps it would be worth writing an integration with it if you're going to use more than just type/token ratio?

```suggestion
            lambda x: len(set(x.str.lower().split(" ")) / len(x.str.split(" "))
```

Reply:
Strong agree here - keeps things more modular, and keeps the scope of timeseriesflattener more focused.